File: examples/hello-world/gabriel-greeting-dart/api/public_test.dart
import 'package:test/test.dart';

import '../gen/dart/greeting/v1/greeting.pb.dart';
import 'public.dart' as public_api;

void main() {
  test('listLanguages includes English', () {
    final response = public_api.listLanguages(ListLanguagesRequest());
    final english = response.languages.firstWhere((language) => language.code == 'en');

    expect(english.name, equals('English'));
    expect(english.native, equals('English'));
  });

  test('sayHello uses requested language', () {
    final response = public_api.sayHello(
      SayHelloRequest()
        ..name = 'Alice'
        ..langCode = 'fr',
    );

    expect(response.greeting, equals('Bonjour Alice'));
    expect(response.language, equals('French'));
    expect(response.langCode, equals('fr'));
  });

  test('sayHello uses localized default name', () {
    final response = public_api.sayHello(
      SayHelloRequest()..langCode = 'ja',
    );

    expect(response.greeting, equals('こんにちは、マリアさん'));
    expect(response.language, equals('Japanese'));
    expect(response.langCode, equals('ja'));
  });

  test('sayHello falls back to English', () {
    final response = public_api.sayHello(
      SayHelloRequest()..langCode = 'unknown',
    );

    expect(response.greeting, equals('Hello Mary'));
    expect(response.language, equals('English'));
    expect(response.langCode, equals('en'));
  });
}

File: examples/hello-world/gabriel-greeting-node/api/public_test.js
'use strict';

const test = require('node:test');
const assert = require('node:assert/strict');

const pb = require('../gen/node/greeting/v1/greeting_pb.js');
const publicApi = require('./public');

test('listLanguages includes English', () => {
  const response = publicApi.listLanguages(new pb.ListLanguagesRequest());
  const english = response.getLanguagesList().find((language) => language.getCode() === 'en');

  assert.ok(english);
  assert.equal(english.getName(), 'English');
  assert.equal(english.getNative(), 'English');
});

test('sayHello uses requested language', () => {
  const request = new pb.SayHelloRequest();
  request.setName('Bob');
  request.setLangCode('fr');

  const response = publicApi.sayHello(request);

  assert.equal(response.getGreeting(), 'Bonjour Bob');
  assert.equal(response.getLanguage(), 'French');
  assert.equal(response.getLangCode(), 'fr');
});

test('sayHello uses localized default name', () => {
  const request = new pb.SayHelloRequest();
  request.setLangCode('ja');

  const response = publicApi.sayHello(request);

  assert.equal(response.getGreeting(), 'こんにちは、マリアさん');
  assert.equal(response.getLanguage(), 'Japanese');
  assert.equal(response.getLangCode(), 'ja');
});

test('sayHello falls back to English', () => {
  const request = new pb.SayHelloRequest();
  request.setLangCode('unknown');

  const response = publicApi.sayHello(request);

  assert.equal(response.getGreeting(), 'Hello Mary');
  assert.equal(response.getLanguage(), 'English');
  assert.equal(response.getLangCode(), 'en');
});
